src/frontend/screens/Sync/ProjectSyncDisplay.tsx

```diff
@@ -99,146 +184,344 @@ export const ProjectSyncDisplay = ({
+          fullWidth
+          variant="outlined"
+          onPress={() => {
+            // TODO: Catch/surface error
+            projectApi.$sync.stop();
+          }}>
+          <View style={styles.buttonContentContainer}>
+            <StopIcon size={20} color={BLACK} />
+            <Text style={styles.buttonTextSecondary}>{t(m.stop)}</Text>
+          </View>
+        </Button>
+      );
+
+      syncInfoContent = (
+        <>
+          <Text style={styles.titleText}>{t(m.waitingForDevices)}</Text>
+          <SyncProgress stage={syncStage} />
+        </>
+      );
+
+      break;
+    }
+    case 'syncing': {
+      dockContent = (
+        <Button
+          fullWidth
+          variant="outlined"
+          onPress={() => {
+            // TODO: Catch/surface error
+            projectApi.$sync.stop();
+          }}>
+          <View style={styles.buttonContentContainer}>
+            <StopIcon size={20} color={BLACK} />
+            <Text style={styles.buttonTextSecondary}>{t(m.stop)}</Text>
+          </View>
+        </Button>
+      );
+
+      syncInfoContent = (
+        <>
+          <Text style={styles.titleText}>
+            {syncStage.progress === 0
+              ? t(m.waitingForDevices)
+              : t(m.syncingWithDevices, {
+                  active: syncingPeersCount,
+                  total: connectedPeersCount,
+                })}
+          </Text>
+          <SyncProgress stage={syncStage} />
+        </>
+      );
+
+      break;
+    }
+    case 'complete-partial': {
+      dockContent = (
+        <Button
+          fullWidth
+          variant="outlined"
+          onPress={() => {
+            // TODO: Catch/surface error
+            projectApi.$sync.stop();
+          }}>
+          <View style={styles.buttonContentContainer}>
+            <StopIcon size={20} color={BLACK} />
+            <Text style={styles.buttonTextSecondary}>{t(m.stop)}</Text>
+          </View>
+        </Button>
+      );
+
+      syncInfoContent = (
+        <>
+          <Text style={styles.titleText}>
+            {t(m.syncingCompleteButWaitingForOthers, {
+              count:
+                syncStage.connectedPeersCount - syncStage.syncingPeersCount,
+            })}
+          </Text>
+          <SyncProgress stage={syncStage} />
+        </>
+      );
+
+      break;
+    }
+    case 'complete-full': {
+      dockContent = (
+        <Button variant="text" disabled onPress={() => {}}>
+          <Text style={styles.buttonTextSecondary}>{t(m.allCaughtUp)}</Text>
+        </Button>
+      );
+
+      syncInfoContent = (
+        <>
+          <View>
+            <Text style={styles.titleText}>{t(m.syncingFullyComplete)}</Text>
+            <Text style={styles.subtitleText}>{t(m.allDataSynced)}</Text>
+          </View>
+          <SyncProgress stage={syncStage} />
+        </>
+      );
+
+      break;
+    }
+    default: {
+      // @ts-expect-error
+      throw new Error(`Invalid status: ${syncStage.status}`);
```

Comment:
Nit: consider [`ExhaustivenessError`][0]. Same comment applies to the `switch` below.

[0]: https://github.com/digidem/comapeo-mobile/blob/981473069a947a81e12990093230186183e605ca/src/frontend/lib/ExhaustivenessError.ts

Reply:
addressed via https://github.com/digidem/comapeo-mobile/pull/619/commits/b11818b855f351a5a76bb2b443a7665b5b172a55